### PR TITLE
JS - Basics Diagnostic complete

### PR DESCRIPTION
**Pull request by akadane87**
Done

### CODE REVIEW

File: bin/diagnostic.js [hunk 1]
```diff
@@ -1 +1,20 @@
...
+
+let answer = 'not empty';
+
+while (answer !== '') {
+  answer = ask("Guess my secret? ");
+  if (answer === 'SeCrEt') {
+   console.log("You guessed it!");
+   break;
+  }
+}
```
Comment (neugierige) [inline bin/diagnostic.js:13]: I'm not sure if you ran out of time on this one, but please look this over again. Here's one possible solution. If this is unclear to you, please let us know.

```
const ask = require('../lib/ask.js');

let answer = '';

while (answer !== '5') {
  if (answer < 1 || answer > 10) {
   console.log('Please try a number between 1 and 10');
 } else if (answer !== '') {
   console.log('That\'s not it!');
 }
  answer = ask("Guess my secret? ");
}
console.log("You guessed it!");
```

File: diagnostic.md [hunk 2]
```diff
@@ -9,6 +9,17 @@ Assume all code executes in strict mode.
 
 Please list the JavaScript primitives and give one example of each.
 
+Boolean : true
+Null : Null
+Undefined : let red = ;
```
Comment (neugierige) [inline diagnostic.md:14]: Undefined is just undefined.

File: diagnostic.md [hunk 3]
```diff
@@ -9,6 +9,17 @@ Assume all code executes in strict mode.
...
+String : ('cat', 'dog',)
```
Comment (neugierige) [inline diagnostic.md:15]: 'cat' and 'dog' are individually examples of string, but the expression you wrote is not.

File: diagnostic.md [hunk 4]
```diff
@@ -9,6 +9,17 @@ Assume all code executes in strict mode.
...
+Array : [num1, num2 num3]
```
Comment (neugierige) [inline diagnostic.md:16]: In an array, there are commas between every two elements. Here, you are missing a comma between num2 and num3.

File: diagnostic.md [hunk 5]
```diff
@@ -9,6 +9,17 @@ Assume all code executes in strict mode.
...
+Number : 7
+Symbol: ('foo')
```
Comment (neugierige) [inline diagnostic.md:18]: Symbol is not a JavaScript primitive.

File: diagnostic.md [hunk 6]
```diff
@@ -9,6 +9,17 @@ Assume all code executes in strict mode.
...
+object : car
```
Comment (neugierige) [inline diagnostic.md:19]: Object is not a JavaScript primitive

**Comment by neugierige** [comment]
3/5. Hi Adam, please review JavaScript primitive types again. Look back at the js-basics repo. I would also encourage you to take another stab at question 5.

**Comment by akadane87** [comment]
I would say in general I'm feeling pretty behind as of today

**Comment by neugierige** [comment]
Hi Adam, so what are you going to do about it? What is your plan of action for catching up on this stuff? Don't panic. Make a plan with specific steps.

**Comment by akadane87** [comment]
Thats the thing Luyuan...I don't have a plan to catch myself up on this
stuff and that's EXACTLY what I have been asking you all for...a plan.  Let
me preface what I'm about to say with I'm totally calm here. I have spoken
to you all individually and all of you have said don't worry - it will come
to you. I am 100% taking that advice. Where I guess I feel a little
frustrated is I'm not sure you guys are hearing me when I say that just
"reviewing the materials" is not working for me. I'm here for 5 extra hours
every day (with the exception of yesterday) 2 in the morning and 3 after
class. Please believe me when I say looking at old repos is not getting me
anywhere at my current level of understanding. My learning style
(unfortunately) is not typical of most - it never has been. If you were to
hand me an article on something it would take me much longer than the
average bear to read and I wouldn't retain all of it. Where I do learn best
is videos, interactive tutorials and 1on1's. So what I'm asking of all you
is some alternative methods to digest this material. I'm asking for help to
come up with a plan to get back on track but it can't be "look at the
materials again" because that is not helping.

**Comment by neugierige** [comment]
Hey Adam, I hear you. First of all, please understand that it's ok if some things don't make sense right now, because you are being introduced to a lot of ideas in the abstract. Once you start applying them (for the game project, for example), you will see the benefit & necessity of these tools and they will make much more sense. This was my experience with programming. 

If the way you have been studying is not working, you need to try something else. Let's chat during 1:1s today. Are you available at 4:30?

**Comment by akadane87** [comment]
Sounds great Lu, thanks very much.

**Pull request closed without merging**